app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala

```diff
@@ -184,6 +184,18 @@ class IfConnector @Inject() (
       )
       Future.failed(new BadRequestException(s"Invalid appDate: $invalidAppDate"))
 
+    case UpstreamErrorResponse(msg, 400, _, _)
+      if requestUrl.contains("/employee/counts") && msg.contains("INVALID_PAYLOAD") =>
```

Comment:
Am wondering if we can make this a generic thing than being specific here. Maybe there is a reason for `/vat` endpoint to be specific to capture the wrong date parameter. but, this should be a general 400 handling that propagates back a BadRequest. Otherwise, another endpoint that has similar issue will still result in a 400 -> 500 translation.